server/tests/api_v4/authenticated/test_block_create.py

```diff
@@ -27,6 +31,28 @@ async def test_authenticated_block_create_ok(coolorg: CoolorgRpcClients, backend
     assert dump == expected_dump
 
 
+async def test_authenticated_block_create_realm_not_found(
+    coolorg: CoolorgRpcClients, backend: Backend
+) -> None:
+    block_id = BlockID.new()
+    block = b"<block content>"
+
+    # Try to create block with an invalid realm ID
+    rep = await coolorg.alice.block_create(block_id, VlobID.new(), block)
+    assert rep == authenticated_cmds.v4.block_create.RepRealmNotFound()
+
+    # Block should not exist
+    content = await backend.block.read_as_user(
```

Comment:
No need for this part given `test_dump_blocks` already shows us the block doesn't exist

Note: `read_as_user` is present for the `_ok`  test case. This is not strictly required, but works as a cheap check to ensure the block can be actually retrieved (while testing the block cannot be retrieved when it is not present in the first place seems very paranoid).

In the end I'm fine with either of those solutions:
- do nothing and keep `read_as_user` check in all tests
- considering `test_dump_blocks` is already enough and remove `read_as_user` in all tests 
- keep `read_as_user` only in the `_ok` test to balance test readability

your call :+1:

Reply:
You're right. Actually, if ever `read_as_user` check fails, but not `test_dump_blocks`, that would mean there is a bug in `read_as_user`, not really in `block_create`.

Anyway, maybe it is better to leave it as a "cheap check". I added some comments based on your review.
WDYT?

Reply:
> WDYT

:+1: